bot/exts/utils/snekbox/_eval.py

```diff
@@ -70,41 +70,66 @@ def status_emoji(self) -> str:
         else:  # Exception
             return ":x:"
 
-    def get_message(self, job: EvalJob) -> tuple[str, str]:
-        """Return a user-friendly message and error corresponding to the process's return code."""
-        msg = f"Your {job.version} {job.name} job has completed with return code {self.returncode}"
+    @property
+    def error_message(self) -> str:
+        """Return an error message corresponding to the process's return code."""
         error = ""
-
         if self.returncode is None:
-            msg = f"Your {job.version} {job.name} job has failed"
             error = self.stdout.strip()
-        elif self.returncode == 128 + SIGKILL:
-            msg = f"Your {job.version} {job.name} job timed out or ran out of memory"
         elif self.returncode == 255:
-            msg = f"Your {job.version} {job.name} job has failed"
             error = "A fatal NsJail error occurred"
+        return error
+
+    @property
+    def files_error_message(self) -> str:
+        """Return an error message corresponding to the failed files."""
+        if not self.failed_files:
+            return ""
+
+        failed_files = f"({self.failed_files_str()})"
+
+        n_failed = len(self.failed_files)
+        files = f"file{'s' if n_failed > 1 else ''}"
+        msg = f"Failed to upload {n_failed} {files} {failed_files}"
+
+        if (n_failed + len(self.files)) > FILE_COUNT_LIMIT:
+            it_they = "they" if n_failed > 1 else "it"
+            msg += f" as {it_they} exceeded the {FILE_COUNT_LIMIT} file limit."
+        else:
+            msg += f". File sizes should each not exceed {sizeof_fmt(FILE_SIZE_LIMIT)}."
+
+        return msg
+
+    def failed_files_str(self, char_max: int = 85, file_max: int = 5) -> str:
```

Comment:
Should be named `get_failed_files_str`, and maybe also be private.

Reply:
Renamed in [d7722b3](https://github.com/python-discord/bot/pull/2326/commits/d7722b3c335af13d43d3b958c34802d2a98d0279)

Not sure if it should be private, it doesn't mutate or allow any access that needs to be protected, maybe invoking it directly could be useful for custom error logging or such?